iceoryx2-pal/concurrency-sync/src/iox_atomic.rs

```diff
@@ -55,30 +55,14 @@ pub type IoxAtomicI16 = core::sync::atomic::AtomicI16;
 #[allow(clippy::disallowed_types)]
 pub type IoxAtomicI32 = core::sync::atomic::AtomicI32;
 
-#[cfg(all(
-    target_pointer_width = "64",
-    not(feature = "enforce_32bit_rwlock_atomic")
-))]
 /// Behaves like [`core::sync::atomic::AtomicI64`]
 #[allow(clippy::disallowed_types)]
 pub type IoxAtomicI64 = core::sync::atomic::AtomicI64;
 
-#[cfg(all(
-    target_pointer_width = "64",
-    not(feature = "enforce_32bit_rwlock_atomic")
-))]
 /// Behaves like [`core::sync::atomic::AtomicU64`]
 #[allow(clippy::disallowed_types)]
 pub type IoxAtomicU64 = core::sync::atomic::AtomicU64;
 
-#[cfg(any(target_pointer_width = "32", feature = "enforce_32bit_rwlock_atomic"))]
-/// Non lock-free implementation that behaves like [`core::sync::atomic::AtomicI64`]
-pub type IoxAtomicI64 = IoxAtomic<i64>;
-
-#[cfg(any(target_pointer_width = "32", feature = "enforce_32bit_rwlock_atomic"))]
-/// Non lock-free implementation that behaves like [`core::sync::atomic::AtomicU64`]
-pub type IoxAtomicU64 = IoxAtomic<u64>;
```

Comment:
What do you think about renaming the `IoxAtomic` to `IoxAtomicWithLock`? The name might fool developers in believing the atomic is lock-free, like guaranteed with the built-in Rust atomics.

Reply:
I called it `IoxAtomic` on purpose. The loom testing library for instance requires its own atomic, see: https://crates.io/crates/loom

Also, we may want to instrument the atomic to achieve 100% mc/dc coverage for compare exchange cases.

So the `IoxAtomic` is from the user point of view just an alias to the actual atomic implementation, which is sometimes adjusted to `loom::sync::Atomic` or `iceoryx2_testing_platform::Atomic` fit other use cases.

The `IoxAtomicWithLock` would be something else in my opionion.